codeit/services.js/ArticleService.js

```diff
@@ -0,0 +1,125 @@
+const BASE_URL = 'https://panda-market-api-crud.vercel.app'; // 기본이 되는 URL
+
+const ArticleService = { 
+    
+// 1. 게시글 목록 조회 :  getArticleList 함수
+    getArticleList(params = {page : 1, pageSize: 10, keyword: ''} ) { // 이 함수를 호출할 때, 어떤 조건으로 보여줄지 기본값& 파라미터를 설정함
+        const url = new URL(`${BASE_URL}/articles`); // 기본 url에 path 주소를 추가함
+
+        Object.keys(params) // 객체로 되어있는 params의 프로퍼티 네임(키)들을 배열로 만듦
+            .forEach((key)=>{ // 배열로 만든 params의 키들을 하나씩 꺼내서 콜백 함수 실행
+                if (params[key] !== undefined && params[key] !== '') { // params[key] 는 해당 키의 값 -> 값이 undefined, 빈 문자열이 아닐경우에만 동작
+                    url.searchParams.append(key, params[key]); // new URL()로 만든 URL 객체 뒤에 파라미터에 들어가 있는 키와 값을 한 쌍의 쿼리 파라미터를 추가해주는 메서드
+                }
+            })
```

Comment:
for ...of와 Object.entries로 가독성을 더 개선해 볼 수 있겠네요.
```suggestion
     for (const [key, value] of Object.entries(params)) {
         if (value !== undefined && value !== "") {
             url.searchParams.set(key, value); // append 대신 set 사용
         }
     }
```